### PR TITLE
Image IO componentType, pixelType options

### PR DESCRIPTION
**Pull request by thewtex**
- `pixelType` and `componentType` options to image IO functions
- Same option object conventions as other pipeline functions. Existing options arguments still supported through backwards compatibility code.

### CODE REVIEW

**Comment by thewtex** [comment]
Closes #671

File: src/io/readImageDICOMArrayBufferSeries.ts [hunk 1]
```diff
@@ -78,11 +80,29 @@ const seriesBlockSize = 8
...
+  let fileNames: undefined | string[]
+  if (typeof options === 'boolean') {
+    // Backwards compatibility
+    singleSortedSeries = options
+  }
+  if (typeof fileNamesBackwardsCompatibility !== 'undefined') {
+    fileNames = fileNamesBackwardsCompatibility
+  }
+  if (typeof options === 'object') {
+    if (typeof options.singleSortedSeries !== 'undefined') {
```
Comment (sankhesh) [inline src/io/readImageDICOMArrayBufferSeries.ts:96]: Should `options.singleSortedSeries` type be enforced as boolean?

**Review by sankhesh — APPROVED** [review]
LGTM overall except that one question

File: src/io/readImageDICOMArrayBufferSeries.ts (hunk 1 above)
Comment (thewtex) [reply src/io/readImageDICOMArrayBufferSeries.ts:96]: Yes, this is already enforced via Typescript at compile time. This helps remove overhead of a run-time check.

**Pull request merged**